Content.Client/Administration/Commands/UploadFolder.cs

```diff
@@ -0,0 +1,83 @@
+using System.IO;
+using System.Resources;
+using System.Threading.Tasks;
+using Content.Shared.Administration;
+using Content.Shared.CCVar;
+using Robust.Client.UserInterface;
+using Robust.Shared.Configuration;
+using Robust.Shared.Console;
+using Robust.Shared.Network;
+using Robust.Shared.Utility;
+using Robust.Client.Utility;
+using Robust.Shared.ContentPack;
+
+namespace Content.Client.Administration.Commands;
+
+public sealed class UploadFolder : IConsoleCommand
+{
+    public string Command => "uploadfolder";
+    public string Description => "Uploads a folder recursively to the server contentDB.";
+    public string Help => $"{Command} [folder in userdata/UploadFolder] ";
+
+    private static readonly ResourcePath BaseUploadFolderPath = new("/UploadFolder");
+
+    public async void Execute(IConsoleShell shell, string argStr, string[] args)
+    {
+        var cfgMan = IoCManager.Resolve<IConfigurationManager>();
+        var resourceMan = IoCManager.Resolve<IResourceManager>();
+        var fileCount = 0;
+
+
+        if (!cfgMan.GetCVar(CCVars.ResourceUploadingEnabled))
+        {
+            shell.WriteError("Network Resource Uploading is currently disabled by the server.");
+            return;
+        }
+
+        if (args.Length != 1)
+        {
+            shell.WriteError($"Wrong number of arguments! \n usage: {Command} [folder in userdata/UploadFolder] ");
+            return;
+        }
+
+        if (args[0].Contains(".."))
+        {
+            shell.WriteError($"Illegal characters found in folder path. Remove '..' from your folder path! ");
+            return; // silent bomb out if the user is trying to escape the UploadFolder dir
+        }
```

Comment:
Pointless. Just let people specify whatever path, it's guaranteed `/data` relative.

Reply:
resolved